### ISSUE TITLE
JS Problems

### ISSUE THREAD

**Issue by etm** [sciapp/gr#177]
The javascript tutorial https://gr-framework.org/tutorials/using_gr_in_javascript.html demonstrates some problems with the js backend:

-----

Animation Example (Example 2 on the webpage) - Grid and Axes are missing: If the drawing order in the example is changed to

```javascript
    gr.clearws();
    gr.setwindow(x[0], x[200], -1, 1);
    gr.polyline(200, x, y);
    gr.textext(0.37, 0.97, "Plot Animation Demo");
    gr.textext(0.5, 0.01, "x");
    gr.textext(0.01, 0.525, "y");
    gr.grid(0.25, 0.25, 0, 0, 2, 2);
    gr.axes(0.25, 0.25, x[0], -1, 2, 2, 0.005);
    gr.axes(0.25, 0.25, x[200], 1, -2, -2, -0.005);
```

everything works as expected. The problem in the example on the webpage above is, that if the polyline occurs after grid/axes, then it overwrites them (it draws in a big whit rectangle instead of transparent.

-----

Datagrid Example (Example 1 on the webpage) - colors and surface are missing. Here the polymarker operation erases everything before (surface, contour). Surface of course also erases everything, contour seems to behave correctly and paint with transparent background. Nonetheless there is IMHO currently no means to produce the expected results in JS for example 1.

-----

TLDR: many javascript functions have the problem that they erase what operations before have drawn.

**Comment by danielkaiser** [comment]
Hi @etm,

the issue should be fixed by 349cdfef4ed1d741aa5781a64be28eded570cad3. If you reload the [tutorials page](https://gr-framework.org/tutorials/using_gr_in_javascript.html) the embedded example should work again and the code from the first example should work as well when using the newest [`gr.js`](https://gr-framework.org/downloads/gr-latest.js)

**Comment by etm** [comment]
Hi @danielkaiser,

perfect. thanks.
